src/auto-updater/index.js

```diff
@@ -88,6 +89,21 @@ function setup (ctx) {
     })
   })
 
+  let progressPercentTimeout = null
+  autoUpdater.on('download-progress', ({ percent, bytesPerSecond }) => {
+    const logDownloadProgress = () => {
+      logger.info(`[updater] download progress is ${percent.toFixed(2)}% at ${bytesPerSecond} bps.`)
+    }
+    // log the percent, but not too often to avoid spamming the logs, but we should
+    // be sure we're logging at what percent any hiccup is occurring.
+    clearTimeout(progressPercentTimeout)
+    if (percent === 100) {
+      logDownloadProgress()
+      return
+    }
+    progressPercentTimeout = setTimeout(logDownloadProgress, 2000)
+  })
```

Comment:
> unless the goal is to capture stalled downloads, is it?
Yes, that's part of it. But also to see download speeds of users so we can infer "oh, slow network.. something happened, and it failed" or "oh, something happened and bps reduced significantly, then they got an error"

> don't rely on time and timeouts (which are always finicky in js)

I disagree with this. 

I think we still need the timeout. Basically, if we do not get another `download-progress` event, we need to have the last one logged, no matter what, or else we won't know at what percentage the download failed.

---

A few requirements I think we should have for this log:

1. Do not spam the log (logging every 5 % is too much.. maybe every 25% is ideal)
2. If download % is 100, we should log immediately
3. If download fails at any point, the last download-progress event should be logged, no matter the value.

The 3rd criteria here is missing from your changes, and part of why I did the setTimeout. No matter what, the last event will be logged when setTimeout resolves. Criteria#3 is also challenging because a signal for "this is the last download-progress event" is missing from electron. 

In a perfect world, electron would emit an actual 'updateFailed' event, and we could store the last percent received and output at that point, but electron only emits an 'error' event, which is also emitted for non-fatal errors (wth?)

One failure with my 2-second timeout is that Criteria#1 is invalidated for slow networks.